server/auth.go

```diff
@@ -374,6 +375,38 @@ func login(user *User, db *gorm.DB) (AuthResponse, error) {
 	return AuthResponse{Token: token}, nil
 }
 
+func loginProxy(user *User, db *gorm.DB) (AuthResponse, error) {
+	slog.Debug("A User Is Logging In", "username", user.Username)
+	dbUser := new(User)
+	res := db.Where("username = ? AND (type IS NULL OR type = 0 OR type = ?)", user.Username, PROXY_USER).Take(&dbUser)
```

Comment:
I solved it this way to allow the following behaviour I already explained above:

> If the user already exists either with no UserType (e.g. username + password) or PROXY_USER they are authenticated.
> If no user exists with that username and either no UserType or PROXY_USER as type then a new user with UserType PROXY_USER is created.

With this method you can migrate an existing setup to forward authentication without issues (as long usernames in watcharr and your auth solution match) and the Admin user created when first launching Watcharr can also be migrated to forward auth.

If you don't like this behavior I can of course change it. But as soon as you implement automatic auth with forward auth, it is needed as otherwise you won't be able to log into the main account anymore.

Reply:
Ah okay I see @lufixSch, that does make more sense, I can understand server admins wanting an easy migration for using their existing accounts with forward auth.

I think it might be possible to break, for example, if someone logs in via proxy and they have a PROXY_USER type, but then they or someone else makes a normal user type with the same username. When the user tries logging in via proxy, it might always select the wrong one?

If that made sense and you agree, I could look into supporting converting user types to PROXY_USER in the admin user dashboard if you think that is a possible solution (or if you have another way around that or think its not a problem, let me know)?

Reply:
You are right this could be an issue. I did not think of that.

The ability to migrate users would be a viable solution. 
Another Idea would be to check for an existing user with `PROXY_USER` type before registering a new user. But that would still mean one could never create a user for regular login using the same name as already used by a forward auth user.

Usually when using proxy authentication, forward auth would be the only login method allowed on the service as the only way to reach the page is through the proxy (As soon as automatic login is implemented a user wouldn't be able to reach the login page anyway). Therefore, I don't think of it as a big issue. This doesn't mean it shouldn't be fixed but that the easiest/fastest solution is probably the best one.